src/app/core/components/header/user-panel/user-panel.component.ts

```diff
@@ -8,6 +14,17 @@ import { faSignOutAlt, faUser, IconDefinition } from '@fortawesome/free-solid-sv
   styleUrls: [ './user-panel.component.scss' ],
 })
 export class UserPanelComponent {
+  @Input() userName$: Observable<string>;
+
   iconUser: IconDefinition = faUser;
   iconSignOut: IconDefinition = faSignOutAlt;
+
+  constructor(private router: Router, private authService: AuthService) { }
+
+  logout() {
+    this.authService.logout$().subscribe(
+      () => this.router.navigate([ AppRoutePath.Login ]),
+      () => alert('An error has occured during the logging out'),
+    );
```

Comment:
1. please use `$` and `$$` only for `get` properties. here is no need to use it for methods.
2. use `take(1),` to complete a stream after first emit. It also exists `first()`, `last()`, `takeUntil`helpers which are pretty useful.
```typescript
    this.authService.logout$().pipe(  
take(1),    
tap(() => this.router.navigate([ AppRoutePath.Login ])),
tap(() => alert('An error has occured during the logging out')),
) .subscribe();
```

Reply:
Okay, thanks, I got general idea. However, in this case seems like error alert should be always shown, even if we don't have error